Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -695,7 +695,18 @@ public extension Purchases {
     ///
     @objc(switchUserToNewAppUserID:)
     func switchUser(to newAppUserID: String) {
+        guard self.identityManager.currentAppUserID != newAppUserID else {
+            self.logger.warn(Strings.identity.switching_user_same_app_user_id(newUserID: newAppUserID))
+            return
+        }
```

Comment:
we missed this in the original implementation but it becomes a lot more important when a cache update is added to the mix

Reply:
👍🏻